doc/scheduled-configurations/scheduled-configurations-hld.md

```diff
@@ -0,0 +1,547 @@
+
+### Subscription APPL_DB
+
+In order to fully integrate with the new scheduled configurations feature, all switch components desiring to utilize this functionality must subscribe to the `APPL_DB`. The table schema will be the same as the schema used in `CONFIG_DB` to apply new configurations, all that is needed is to add the new subscription to the orchagent subcomponents. This is done in order to distinguish between administrator configurations, and configurations applied due to an application (ie. *Scheduled Configurations*).
+
+The following diagram shows the connections between the different components:
+
+![Architecture Design](images/scheduled-configuration-architecture.png)
+
+### Flow Diagrams
+
+The following diagrams illustrate the interactions between the databases and applications when setting the time range and scheduled configurations settings, as well as what happens when an event is triggered.
+
+#### Set
+
+![time range flow -- set](images/time-range-flow-set.png)
+
+1. Apply time range configuration to `TIME_RANGE` in `CONFIG_DB` through sonic-cfggen or other management tool
+2. Changes to the `TIME_RANGE` table will cause a subscription notification to be sent to `timerangemgrd`
+3. `timerangemgrd` will create a new entry in the `TIME_RANGE_STATUS_TABLE` table found in the `STATE_DB` defaulting to *active* or *inactive* status based on the current time
+4. `timerangemgrd` will then create the crontab files used by the `cron` application to activate, or deactivate the status of the `TIME_RANGE_STATUS_TABLE`
+
+![scheduled configuration flow -- set](images/scheduled-configuration-flow-set.png)
+
+1. Apply scheduled configurations to the `SCHEDULED_CONFIGURATIONS` table in `CONFIG_DB`
+2. Changes to `SCHEDULED_CONFIGURATIONS` will cause a subscription notification to `scheduledconfigmgrd`
+3. If the time range this configuration is already active, it will publish this new configuration to the `APPL_DB`
+4. `scheduledconfigmgrd` saves the configuration internally
+
+#### Delete
+
+![time range flow -- delete](images/time-range-flow-del.png)
+
+1. A time range is deleted from the `TIME_RANGE` table in `CONFIG_DB`
+2. A subscription notification is sent to `timerangemgrd` with the op *delete*
+3. `timerangemgrd` deletes the crontab files associated with this time range
+4. `timerangemgrd` deletes the time range entry from the `TIME_RANGE_STATUS_TABLE` table found in the `STATE_DB`
+5. A subscription notification is sent to `scheduledconfigmgrd` the op *delete*
+6. `scheduledconfigmgrd` checks if the time range is active, if yes then it will deactivate the configurations bound to this time range by deleting the configurations from the `APPL_DB`
+7. Finally, `scheduledconfigmgrd` deletes the time range internally, and the scheduled configurations will be unbound
```

Comment:
what if the there were some old configuration being replaced by this scheduled configuration, are we recovering old configuration? 

also, what if scheduled configuration is changed, how do we revert?

Reply:
the feature was designed to apply an additional configuration rather than changing exiting configuration, this conceptionally remove the need to "restore" old configuration, handle conflicting configurations (e.g. user change the default configuration while in scheduled configuration).
i think it should be the system admin responsibility to handle conflicts, as the scheduled configuration is a configuration and the default configuration is also configuration, two options can be used :
one all scheduled configuration that is conflicting should be placed as scheduled configuration, e.g. assume "new" configuration in time range X so need to set "default" configuration to time range ~X ( already supported)
second option : on time range end allow admin to set configuration ( rather then just delete configuration), this allow admin to reconfigure the "default"
regrading the second question on time configuration change need to remove the scheduled configuration and configure a new one

Reply:
@lguohan How do these options in the design sound to you?